### PR TITLE
Round-trip time calculation with RTCP sender and receiver reports

### PR DESCRIPTION
**Pull request by zhuker**
Is this the best place for it?

### CODE REVIEW

File: content/docs/06-media-communication.md [hunk 1]
```diff
@@ -200,12 +200,36 @@ The first road block with implementing Congestion Control is that UDP and RTP do
...
+
+Sender and Receiver reports (SR and RR) work together to compute round-trip time.
+
+Sender includes its local time `sendertime1` in SR.
+When receiver gets SR packet it sends back RR.
+Among other things RR includes `sendertime1` just received from sender.
+There's a delay between receiving SR and sending RR, because of that RR also includes "delay since last sender report" time `DLSR`. 
+`DLSR` is used to adjust round-trip time estimate later on in the process.
+Once sender receives RR it subtracts `sendertime1` and `DLSR` from current time `sendertime2`.
+This time delta is called round-trip propagation delay or round-trip time.
```
Comment (mogren) [inline content/docs/06-media-communication.md:219]:
```
The sender includes its local time, `sendertime1` in SR. When the receiver gets the SR packet,
it sends back an RR. Among other things, the RR includes the `sendertime1` just received from sender.
There will be a delay between receiving the SR and sending the RR. Because of that, the RR also
includes a "delay since last sender report" time ,`DLSR`. The `DLSR` is used to adjust the round-trip
time estimate later on in the process. Once the sender receives the RR, it subtracts `sendertime1` and
`DLSR` from the current time, `sendertime2`. This time delta is called the round-trip propagation delay,
or round-trip time.
```

File: content/docs/06-media-communication.md [hunk 2]
```diff
@@ -200,12 +200,36 @@ The first road block with implementing Congestion Control is that UDP and RTP do
...
+`rtt = sendertime2 - sendertime1 - DLSR`
+
+Round-trip time in plain English: 
+- I send you a message with my clock's reading (say it is 4:20pm 42seconds and 420 milliseconds). 
+- You send me this very time back. 
+- You also include the time it took you from reading my message to sending the message back (5 milliseconds).
+- Once I receive the time back I look at the clock again.
+- Now my clock says 4:20pm 42 seconds 690 milliseconds
+- It means that it took 265 milliseconds (690 - 420 - 5) to reach you and return back to me.
+- Therefore round-trip time is 265 msec.
```
Comment (mogren) [inline content/docs/06-media-communication.md:230]: Thanks, this is super helpful. Some suggestions:
```
* I send you a message with my clock's current reading, say it is 4:20 pm, 42 seconds and 420 milliseconds. 
* You send me this same timestamp back. 
* You also include the time it took you to read my message and sending it back, say 5 milliseconds.
* Once I receive the timestamp back, I look at the clock again.
* Now my clock says 4:20 pm, 42 seconds and 690 milliseconds!
* It means that it took 265 milliseconds (690 - 420 - 5) for the message to reach you and be returned back to me.
* Therefore, the round-trip time is 265 milliseconds.
```

**Review by mogren — COMMENTED** [review]
I love it!

I think most other images use a white background, but I'm not sure if they should. That can all be adjusted later if needed anyway. Thanks again for correcting the RFC reference and adding these details! 😄

File: content/docs/06-media-communication.md (hunk 1 above)
Comment (zhuker) [reply content/docs/06-media-communication.md:219]: @mogren is there a difference other than removed line feeds?

i thought i would put each sentence into it's own line so it's easier for you to comment

Comment (zhuker) [reply content/docs/06-media-communication.md:219]: never mind - i see commas and typos

File: content/docs/06-media-communication.md (hunk 2 above)
Comment (zhuker) [reply content/docs/06-media-communication.md:230]: > took you to read my message and sending it back
this sounds ambiguous
i wanted to  say that it is explicitly the time between two events: 
1. receive senders timestamp (420msec)
2. sent the message to the network interface
that is - it does not include time the message took to be actually sent

**Comment by zhuker** [comment]
> I think most other images use a white background

tweaked

File: content/docs/06-media-communication.md [hunk 3]
```diff
@@ -209,12 +209,36 @@ The first road block with implementing Congestion Control is that UDP and RTP do
...
+Among other things, the RR includes `sendertime1` just received from sender.
+There will be a delay between receiving the SR and sending the RR. Because of that, the RR also includes a "delay since last sender report" time - `DLSR`. 
+The `DLSR` is used to adjust the round-trip time estimate later on in the process.
+Once the sender receives the RR it subtracts `sendertime1` and `DLSR` from current time `sendertime2`.
+This time delta is called round-trip propagation delay or round-trip time.
+
+`rtt = sendertime2 - sendertime1 - DLSR`
+
+Round-trip time in plain English: 
+- I send you a message with my clock's current reading, say it is 4:20pm, 42seconds and 420 milliseconds.
```
Comment (mogren) [inline content/docs/06-media-communication.md:233]: 42 seconds

File: content/docs/06-media-communication.md [hunk 4]
```diff
@@ -209,12 +209,36 @@ The first road block with implementing Congestion Control is that UDP and RTP do
...
+This time delta is called round-trip propagation delay or round-trip time.
+
+`rtt = sendertime2 - sendertime1 - DLSR`
+
+Round-trip time in plain English: 
+- I send you a message with my clock's current reading, say it is 4:20pm, 42seconds and 420 milliseconds. 
+- You send me this same timestamp back. 
+- You also include the time elapsed from reading my message to sending the message back, say 5 milliseconds.
+- Once I receive the time back, I look at the clock again.
+- Now my clock says 4:20pm, 42 seconds 690 milliseconds
```
Comment (mogren) [inline content/docs/06-media-communication.md:237]: milliseconds.

File: content/docs/06-media-communication.md [hunk 5]
```diff
@@ -209,12 +209,36 @@ The first road block with implementing Congestion Control is that UDP and RTP do
...
+- It means that it took 265 milliseconds (690 - 420 - 5) to reach you and return back to me.
+- Therefore, the round-trip time is 265 msec.
```
Comment (mogren) [inline content/docs/06-media-communication.md:239]: ..is 265 milliseconds.

**Review by mogren — APPROVED** [review]
Sorry for the nitpicks, just trying to make the text consistent. These sections were definitely needed, thanks a lot for adding it!
